tests/classes/form/validation/FormValidatorControlledVocabTest.php

```diff
@@ -3,86 +3,83 @@
-        $mockControlledVocab->setAssocId(333);
-        $mockControlledVocab->setSymbolic('testVocab');
+        $assocId = (DB::table("publications")
+            ->select("publication_id as id")
+            ->orderBy("publication_id", "desc")
+            ->first()
+            ->id ?? 0) + 100;
 
-        // Set up the mock enumerate() method
-        $mockControlledVocab->expects($this->any())
-            ->method('enumerate')
-            ->willReturn([1 => 'vocab1', 2 => 'vocab2']);
+        $testControlledVocab = Repo::controlledVocab()->build(
+            ControlledVocab::CONTROLLED_VOCAB_SUBMISSION_KEYWORD,
+            Application::ASSOC_TYPE_CITATION,
+            $assocId
+        );
 
-        // Mock the ControlledVocabDAO
-        $mockControlledVocabDao = $this->getMockBuilder(ControlledVocabDAO::class)
-            ->onlyMethods(['getBySymbolic'])
-            ->getMock();
+        $controlledVocabEntryId1 = ControlledVocabEntry::create([
+            'controlledVocabId' => $testControlledVocab->id,
+            ControlledVocab::CONTROLLED_VOCAB_SUBMISSION_KEYWORD => [
+                'en' => 'testEntry',
+            ],
+        ])->id;
 
-        // Set up the mock getBySymbolic() method
-        $mockControlledVocabDao->expects($this->any())
-            ->method('getBySymbolic')
-            ->with('testVocab', Application::ASSOC_TYPE_CITATION, 333)
-            ->willReturn($mockControlledVocab);
-
-        DAORegistry::registerDAO('ControlledVocabDAO', $mockControlledVocabDao);
+        $controlledVocabEntryId2 = ControlledVocabEntry::create([
+            'controlledVocabId' => $testControlledVocab->id,
+            ControlledVocab::CONTROLLED_VOCAB_SUBMISSION_KEYWORD => [
+                'en' => 'testEntry',
+            ],
+        ])->id;
 
         // Instantiate validator
-        $validator = new \PKP\form\validation\FormValidatorControlledVocab($form, 'testData', FormValidator::FORM_VALIDATOR_REQUIRED_VALUE, 'some.message.key', 'testVocab', Application::ASSOC_TYPE_CITATION, 333);
+        $validator = new FormValidatorControlledVocab(
+            $form,
+            'testData',
+            FormValidator::FORM_VALIDATOR_REQUIRED_VALUE,
+            'some.message.key',
+            ControlledVocab::CONTROLLED_VOCAB_SUBMISSION_KEYWORD,
+            Application::ASSOC_TYPE_CITATION,
+            $assocId
+        );
 
-        $form->setData('testData', '1');
+        $form->setData('testData', $controlledVocabEntryId1);
         self::assertTrue($validator->isValid());
 
-        $form->setData('testData', '2');
+        $form->setData('testData', $controlledVocabEntryId2);
         self::assertTrue($validator->isValid());
 
-        $form->setData('testData', '3');
+        $form->setData('testData', 3);
         self::assertFalse($validator->isValid());
+
+        // Delete the test vocab along with entries
+        ControlledVocab::find($testControlledVocab->id)->delete();
```

Comment:
With implementation of setting table based eloquent, we can not add any arbitrary setting data so only options seems to create data and delete at the end of test . Another option can be is to backup/restore specific tables or full DB but that make the test time high.